website/client/js/controllers/groupsCtrl.js

```diff
@@ -71,6 +71,35 @@ habitrpg.controller("GroupsCtrl", ['$scope', '$rootScope', 'Shared', 'Groups', '
       }
     };
 
+    $scope.exportAllMessages = function(messages, timezoneOffset) {
+      
+      if(confirm(env.t('exportInboxMessages'))) {
+
+        var arr = Object.keys(messages).map(function (key) {return messages[key]});
+        
+        // Sort array in descending order
+        arr = _.sortBy(arr, function(num) {
+          return num.sort * -1;
+        });
+
+        var messages = '<html><head></head><body>';
+          for (var i = 0; i < arr.length; i++) {
+            messages += '<p><strong>' + arr[i].user + '</strong> ' + 
+              moment.utc(arr[i].timestamp).zone(timezoneOffset).format('MM/DD/YYYY HH:mm:ss') + 
+              ' (' + (i + 1) + '/' + arr.length + ')<br />' + arr[i].text + '</p>';     
+          }
+          
+        messages += '</body></html>';
+        
+        var blob = new Blob([messages], {type: "text/html"});
```

Comment:
Backwards compatibility or doing this on the backend didn't really even cross my mind, but it makes perfect sense. I'll be moving this to server side soon. It should be implemented in "website/server/controllers/top-level/dataexport.js", right?